databricks/koalas/frame.py

```diff
@@ -2836,6 +2836,71 @@ def merge(left, right: 'DataFrame', how: str = 'inner',
 
         return DataFrame(joined_table.select(*exprs))
 
+    def append(self, other: 'DataFrame', ignore_index: bool = False,
+               verify_integrity: bool = False, sort: bool = False) -> 'DataFrame':
+        """
+        Append rows of other to the end of caller, returning a new object.
+
+        Columns in other that are not in the caller are added as new columns.
+
+        Parameters
+        ----------
+        other : DataFrame or Series/dict-like object, or list of these
+            The data to append.
+
+        ignore_index : boolean, default False
+            If True, do not use the index labels.
+
+        verify_integrity : boolean, default False
+            If True, raise ValueError on creating index with duplicates.
+
+        sort : boolean, default False
+            Currently not supported.
+
+        Returns
+        -------
+        appended : DataFrame
+
+        Examples
+        --------
+        >>> df = ks.DataFrame([[1, 2], [3, 4]], columns=list('AB'))
+
+        >>> df.append(df)
+           A  B
+        0  1  2
+        1  3  4
+        0  1  2
+        1  3  4
+
+        >>> df.append(df, ignore_index=True)
+           A  B
+        0  1  2
+        1  3  4
+        2  1  2
+        3  3  4
+        """
+        if isinstance(other, ks.Series):
+            raise ValueError("DataFrames.append() does not support appending Series to DataFrames")
+        if sort:
+            raise ValueError("The 'sort' parameter is currently not supported")
+
+        index_columns = self._metadata.index_columns
+        if len(index_columns) != len(other._metadata.index_columns):
+            raise ValueError("Both DataFrames have to have the same number of index levels")
+        # TODO Maybe extend to multilevel indices in the future
+        if len(index_columns) != 1 or len(other._metadata.index_columns) != 1:
+            raise ValueError("DataFrames.append() only supports indices with level 1 right now")
```

Comment:
Is the reason why we don't support multilevel index for the following `verify_integrity` check?

Reply:
We can leave it for now anyway.

Reply:
No, it's mainly because I wanted to start with the more simple use case of single-index DataFrames. However, since the existing implementation of `concat()` already supports multi-index DataFrames, I changed `append()` with 79d0359 to support multiple indices as well (as long as they have the same number of levels).